src/utils/links.ts

```diff
@@ -37,11 +37,8 @@ export const getDatasetExplorerLink = (datasetId: string): string => {
 
 /**
  * TODO: full implementation awaiting download-car, parse to file, client-side logic.
- * @see https://github.com/filecoin-project/filecoin-pin-website/issues/29#issuecomment-3403015452
+ * @see https://github.com/filecoin-project/filecoin-pin-website/issues/24
  */
 export const getSpCarDownloadLink = (ipfsRootCid: string, serviceUrl: string, _fileName?: string): string => {
-  // providerInfo.getDownloadLink or from filecoin-pin
-  // download and filename query params are broken currently
-  // return `https://${providerInfo.serviceUrl}/ipfs/${ipfsRootCid}?download=true&filename=${fileName}`
-  return `${serviceUrl}ipfs/${ipfsRootCid}`
+  return `${serviceUrl}ipfs/${ipfsRootCid}?filename=${_fileName}.car`
```

Comment:
@SgtPooki : can you handle the optional string appropriately?  Ideally it isn't optional.

Reply:
fixed with [`9577eaf` (#70)](https://github.com/filecoin-project/filecoin-pin-website/pull/70/commits/9577eaf1bf775d98d6ed0c873afa3167ff90458a)